purchase_stock_picking_show_currency_rate/views/stock_picking_view.xml

```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!--
+License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+-->
+<odoo>
+    <record id="stock_picking_form" model="ir.ui.view">
+        <field name="name">stock.picking.form</field>
+        <field name="model">stock.picking</field>
+        <field name="inherit_id" ref="stock.view_picking_form" />
+        <field name="arch" type="xml">
+            <field name="origin" position="after">
+                <field name="purchase_id" invisible="1" />
+                <field
+                    name="purchase_currency_id"
```

Comment:
Put both in the same line with class `oe_inline` for better display. Check other examples in core.